code/modules/hydroponics/compost_bin.dm

```diff
@@ -0,0 +1,196 @@
+#define BASE_BIOMASS_CAPACITY 1500
+#define BASE_COMPOST_CAPACITY 1500
+#define SOIL_COST 25
+#define SOIL /obj/item/stack/sheet/soil
+#define DECAY 0.2
+#define MIN_CONVERSION 10
+/**
+  * # compost bin
+  * used to make soil from plants.
+  * Doesn't have components.
+  */
+/obj/machinery/compost_bin
+	name = "compost bin"
+	desc = "A wooden bin for composting."
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "compost_bin-empty"
+	power_state = NO_POWER_USE
+	density = TRUE
+	anchored = TRUE
+	/// amount of biomass in the compost bin
+	var/biomass = 0
+	/// amount of compost in the compost bin
+	var/compost = 0
+	/// The maximum amount of biomass the compost bin can store.
+	var/biomass_capacity = BASE_BIOMASS_CAPACITY
+	/// The maximum amount of compost the compost bin can store.
+	var/compost_capacity = BASE_COMPOST_CAPACITY
+	var/composting = FALSE
+
+/obj/machinery/compost_bin/Initialize(mapload)
+	/// try to compost
+	compost()
+	return
```

Comment:
Do we need to try and compost if it spawns empty?

Reply:
I left it there in case it doesn't for any reason. say if some map designer decides to make one with a bit of biomass somewhere. I do think it's relevant because with a small amount of biomass it can take a while to make usable amounts of compost, so there could conceivably be a reason to make one such at roundstart. 
For what little performance this might cost I think it's reasonable.

Reply:
Honestly, that's a mapper issue, if they want they can just edit it to have biomass anyways.
And honestly, while I do see what you mean, in 99% of these cases mappers will never use them.